packages/scandipwa/src/component/MyAccountNewsletterSubscription/MyAccountNewsletterSubscription.container.js

```diff
@@ -81,12 +81,13 @@ export class MyAccountNewsletterSubscriptionContainer extends PureComponent {
             showSuccessNotification,
             newsletterConfirmStatus
         } = this.props;
+        const { isSubscriptionSelected } = this.state;
 
         if (!isSubscribed && wasSubscribed) {
             showSuccessNotification(__('We have removed your newsletter subscription.'));
         } else if (isSubscribed && !newsletterConfirmStatus && !wasSubscribed) {
             showSuccessNotification(__('We have saved your subscription'));
-        } else if (!isSubscribed && newsletterConfirmStatus) {
+        } else if (!isSubscribed && newsletterConfirmStatus && isSubscriptionSelected) {
```

Comment:
Please, check if this logic can be rewritten with checking only 3 parameters instead of 4.
In theory, there should be 1 boolean parameter showing whether email confirmation is required (`newsletterConfirmStatus`), 1 parameter showing current user subscription status saved in DB (`wasSubscribed`?) and 1 showing new selected value (`isSubscriptionSelected`/ `isSubscribed`).

Plus it would be have to leave a comment why each if statement is needed.